src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -155,7 +155,7 @@ BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input,
                 if (input.Type == type)
                 {
                     return _factory.Sequence(
-                        _factory.AssignmentExpression(_factory.Local(target), input),
+                        _factory.AssignmentExpression(target, input),
```

Comment:
It isn't used that way. It is expected to be a fully lowered lvalue.

---
In reply to: [77395353](https://github.com/dotnet/roslyn/pull/13535#discussion_r77395353) [](ancestors = 77395353)